templates/app/controllers/autocomplete_results_controller.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+class AutocompleteResultsController < StoreController
+  def index
+    respond_to do |format|
+      format.html { redirect_to products_path(keywords: params[:keywords]) }
+      format.turbo_stream { load_results }
+    end
+  end
+
+  private
+
+  def load_results
+    @results ||= begin
+      results = {}
+      results[:products] = autocomplete_products
+      results[:taxons] = autocomplete_taxons
+      results
+    end
+  end
+
+  def autocomplete_products
+    if params[:keywords].present?
+      searcher = build_searcher(params.merge(per_page: 5))
+      searcher.retrieve_products
+    else
+      Spree::Product.none
+    end
+  end
+
+  def autocomplete_taxons
+    if params[:keywords].present?
+      Spree::Taxon
+        .where(Spree::Taxon.arel_table[:name].matches("%#{params[:keywords]}%"))
+        .limit(5)
```

Comment:
Non-blocking: the search implementation in `autocomplete_taxons` is on a lower-level compared to `autocomplete_products`. Perhaps `build_searcher` can be updated to handle this query as well?

Reply:
@gsmendoza Yes, that's a good idea, even if a bit out of scope for this PR. At the moment the searcher class used by `build_searcher` assumes you are [searching for a Product](https://github.com/solidusio/solidus/blob/d7ef0f23164a1e5821e4da1cd316d39c6ea96f79/core/lib/spree/core/search/base.rb#L36) and changing this would require a big refactor.